src/katsdpmodels/band_mask.py

```diff
@@ -0,0 +1,102 @@
+################################################################################
+# Copyright (c) 2020, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Masks for roll-off at the edges of a band."""
+
+from typing import Type, TypeVar, ClassVar, Any
+from typing_extensions import Literal
+
+import numpy as np
+import astropy.units as u
+import astropy.table
+import h5py
+
+from . import models
+
+_B = TypeVar('_B', bound='BandMaskRanges')
+
+
+class Band:
+    """Defines a band for use with :class:`BandMask`."""
+
+    def __init__(self, bandwidth: u.Quantity, centre_frequency: u.Quantity):
+        self.bandwidth = bandwidth.to(u.Hz)
+        self.centre_frequency = centre_frequency.to(u.Hz)
+
+    @property
+    def min_frequency(self) -> u.Quantity:
+        return self.centre_frequency - self.bandwidth / 2
+
+    @property
+    def max_frequency(self) -> u.Quantity:
+        return self.centre_frequency + self.bandwidth / 2
+
+
+class BandMask(models.SimpleHDF5Model):
+    model_type: ClassVar[Literal['band_mask']] = 'band_mask'
+
+    def is_masked(self, band: Band, frequency: u.Quantity) -> Any:
+        raise NotImplementedError()      # pragma: nocover
+
+    @classmethod
+    def from_hdf5(cls, hdf5: h5py.File) -> 'BandMask':
+        model_format = models.get_hdf5_attr(hdf5.attrs, 'model_format', str) or ''
+        if model_format == 'ranges':
+            return BandMaskRanges.from_hdf5(hdf5)
+        else:
+            raise models.ModelFormatError(
+                f'Unknown model_format {model_format!r} for {cls.model_type}')
+
+    def to_hdf5(self, hdf5: h5py.File) -> None:
+        raise NotImplementedError()      # pragma: nocover
+
+
+class BandMaskRanges(BandMask):
```

Comment:
I think it would be very helpful to have a docstring describing the broad outline of the model_format in this class. (just the cols, rows and units would be enough). I know this can be parsed out from the code, but it would be nice to have a human reference somewhere.

Reply:
My plan is that once things have stabilised I'll migrate the details from the Google Doc into a manual - but I don't want to duplicate it now while the Google Doc is still being edited and commented on.

Reply:
Fair enough